BaragonData/src/main/java/com/hubspot/baragon/data/BaragonStateDatastore.java

```diff
@@ -225,6 +222,27 @@ public void incrementStateVersion() {
     return serviceStates;
   }
 
+  private Collection<String> getAllServices() {
+    Collection<String> services = new ArrayList<>();
+
+    for (String service : getServices()) {
+      services.add(ZKPaths.makePath(SERVICES_FORMAT, service));
+    }
+
+    return services;
+  }
+
+  public Collection<UpstreamInfo> getAllUpstreams() throws Exception {
```

Comment:
This is a fairly expensive call in terms of zk operations. Do we have the full `Collection<BaragonServiceState>` cached anywhere already (i.e. precomputed). I think there is an in-memory one that is served by the state endpoint. If possible, we should use the in-memory one instead